share/github-backup-utils/ghe-restore-mysql

```diff
@@ -33,13 +33,32 @@ cleanup() {
 }
 trap 'cleanup' INT TERM EXIT
 
+# Check if the backup is binary by looking up the sentinel file 
+is_binary_backup(){
+  test -f $snapshot_dir/mysql-binary-backup-sentinel 
+}
+
+# if mysql.backup.binary feature flag is on
+is_binary_backup_feature_on(){
+  ghe-ssh "$GHE_HOSTNAME" ghe-config --true "mysql.backup.binary"
+}
+
+if is_binary_backup; then
+  if ! is_binary_backup_feature_on; then
+    echo "To restore from a binary backup, you have to set ghe-config \"mysql.backup.binary\" to true" >&2  
+    exit 2
+  fi
+  IMPORT_MYSQL=ghe-import-mysql-xtrabackup
+else
+  IMPORT_MYSQL=ghe-import-mysql-mysqldump
```

Comment:
oops.. Seems this will break back-compat since legacy ghes won't have `ghe-import-mysql-mysqldump` / `ghe-import-mysql-xtrabackup`. We need to stick with `ghe-import-mysql`

Reply:
Fixed with extra checking for feature flag